config/v1/types_cluster_version.go

```diff
@@ -505,6 +505,62 @@ type PromQLClusterCondition struct {
 	PromQL string `json:"promql"`
 }
 
+// ClusterStatusConditionType is an aspect of operator state.
+type ClusterVersionStatusConditionType string
+
+const (
+	// Available indicates that the component (operator and all configured operands)
+	// is functional and available in the cluster. Available=False means at least
+	// part of the component is non-functional, and that the condition requires
+	// immediate administrator intervention.
+	OperatorAvailable ClusterVersionStatusConditionType = "Available"
+
+	// Progressing indicates that the component (operator and all configured operands)
+	// is actively rolling out new code, propagating config changes, or otherwise
+	// moving from one steady state to another. Operators should not report
+	// progressing when they are reconciling (without action) a previously known
+	// state. If the observed cluster state has changed and the component is
+	// reacting to it (scaling up for instance), Progressing should become true
+	// since it is moving from one steady state to another.
+	OperatorProgressing ClusterVersionStatusConditionType = "Progressing"
+
+	// Degraded indicates that the component (operator and all configured operands)
+	// does not match its desired state over a period of time resulting in a lower
+	// quality of service. The period of time may vary by component, but a Degraded
+	// state represents persistent observation of a condition. As a result, a
+	// component should not oscillate in and out of Degraded state. A component may
+	// be Available even if its degraded. For example, a component may desire 3
+	// running pods, but 1 pod is crash-looping. The component is Available but
+	// Degraded because it may have a lower quality of service. A component may be
+	// Progressing but not Degraded because the transition from one state to
+	// another does not persist over a long enough period to report Degraded. A
+	// component should not report Degraded during the course of a normal upgrade.
+	// A component may report Degraded in response to a persistent infrastructure
+	// failure that requires eventual administrator intervention.  For example, if
+	// a control plane host is unhealthy and must be replaced. A component should
+	// report Degraded if unexpected errors occur over a period, but the
+	// expectation is that all unexpected errors are handled as operators mature.
+	OperatorDegraded ClusterVersionStatusConditionType = "Degraded"
```

Comment:
ClusterVersion doesn't do `Degraded` (we use `Failing` with the same semantics), so you can probably drop this and/or recycle its docs for the `Failing` constant.